packages/chain-adapters/src/solana/SolanaChainAdapter.ts

```diff
@@ -602,6 +580,50 @@ export class ChainAdapter implements IChainAdapter<KnownChainIds.SolanaMainnet>
     return instructions
   }
 
+  public async buildEstimationInstructions({
```

Comment:
Leveraging this extracted logic in both `this.buildEstimationSerializedTx`  and send send `estimateFees()`